java/query/query-lambda/src/main/java/sleeper/query/lambda/WarmQueryExecutorLambda.java

```diff
@@ -0,0 +1,156 @@
+import static sleeper.configuration.properties.table.TableProperty.TABLE_NAME;
+import static sleeper.query.runner.output.NoResultsOutput.NO_RESULTS_OUTPUT;
+
+/**
+ * A Lambda that is triggered when an {@link ScheduledEvent} is received. A processor executes, creates a new
+ * {@link Query} for every table in the system and publishes a {@link SendMessageRequest} to the SQS query queue.
+ * The results from the queries are discarded as they are not required since this Lambda ensures that the query Lambdas
+ * remain warm for genuine queries.
+ */
+@SuppressWarnings("unused")
+public class WarmQueryExecutorLambda implements RequestHandler<ScheduledEvent, Void> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(WarmQueryExecutorLambda.class);
+
+    private Instant lastUpdateTime;
+    private InstanceProperties instanceProperties;
+    private TablePropertiesProvider tablePropertiesProvider;
+    private final AmazonSQS sqsClient;
+    private final AmazonS3 s3Client;
+    private final AmazonDynamoDB dynamoClient;
+    private QueryMessageHandler messageHandler;
+    private SqsQueryProcessor processor;
+
+    public WarmQueryExecutorLambda() throws ObjectFactoryException {
+        this(AmazonS3ClientBuilder.defaultClient(), AmazonSQSClientBuilder.defaultClient(),
+                AmazonDynamoDBClientBuilder.defaultClient(), System.getenv(CONFIG_BUCKET.toEnvironmentVariable()));
+    }
+
+    public WarmQueryExecutorLambda(AmazonS3 s3Client, AmazonSQS sqsClient, AmazonDynamoDB dynamoClient, String configBucket) throws ObjectFactoryException {
+        this.s3Client = s3Client;
+        this.sqsClient = sqsClient;
+        this.dynamoClient = dynamoClient;
+        instanceProperties = loadInstanceProperties(s3Client, configBucket);
+        tablePropertiesProvider = new TablePropertiesProvider(instanceProperties, s3Client, dynamoClient);
+    }
+
+    @Override
+    public Void handleRequest(ScheduledEvent event, Context context) {
+        LOGGER.info("Starting to build queries for the tables");
+        S3TableProperties.getStore(instanceProperties, s3Client, dynamoClient)
+                .streamAllTables()
+                .forEach(tableProperty -> {
+                    Schema schema = tableProperty.getSchema();
+                    Field field = schema.getRowKeyFields().get(0);
+
+                    // Create the max and min values for the query. We don't care what they are as long as the query runs
+                    Type type = field.getType();
+                    Object min, max;
+                    if (type instanceof IntType) {
+                        min = 0;
+                        max = 1;
+                    } else if (type instanceof LongType) {
+                        min = 0;
+                        max = 1;
+                    } else if (type instanceof StringType) {
+                        min = "a";
+                        max = "aa";
```

Comment:
I wonder if there's a case where this could include quite a lot of data?

Reply:
You could make this an exact query, where the min and max are the same (like in the python API). That should ensure only 1 record is returned

Reply:
I have set the max and min to the same value as I believe that will have the same effect. I have deployed and tested and it still works without any issues.